src/Elastic.Apm/Agent.cs

```diff
@@ -94,13 +94,14 @@ public static class Agent
 		private static readonly Lazy<ApmAgent> Lazy = new Lazy<ApmAgent>(() => new ApmAgent(_components));
 		private static AgentComponents _components;
 
-
-		public static IConfigurationReader Config => Lazy.Value.ConfigurationReader;
+		public static IConfigurationReader Config => Instance.ConfigurationReader;
 
 		internal static ApmAgent Instance => Lazy.Value;
 
 		internal static bool IsInstanceCreated => Lazy.IsValueCreated;
 
+		public static bool IsConfigured { get; private set; }
```

Comment:
Isn't `IsInstanceCreated` basically the same? 🤔  - that is currently `internal`, so can't be used, but maybe we could just make it `public` and reuse that.

Also another thing: It'd be nice to have some unit tests for this.

Reply:
Note quite the same. Similar to how I answered [here](https://stackoverflow.com/questions/58916834/#comment104099777_58917376) for the example code in that SO question, `IsConfigured` only returns `true` if `Setup` (the configuration method) has been called, while `IsInstanceCreated` would still return `false` because `Lazy` has not yet been evaluated. `IsInstanceCreated` will only return `true` after `Lazy` has been evaluated such as after `Agent.Tracer` has been called.

That said, having looked at how `IsInstanceCreated` is used in the code, I see no value in it versus `IsConfigured` whose logic is equally appropriate for all usages. Therefore, in my latest commit, I've removed `IsInstanceCreated` (it's internal, so not a breaking change) in favour of `IsConfigured` which is for public use. `IsInstanceCreated` was used by some internal tests (via `AgentSingletonUtils.EnsureInstanceCreated`), but these work fine without it (the test itself is unaffected).

Note that the solution wouldn't build locally because `AspNetFullFrameworkSampleApp` references `..\..\packages` which is not populated by any project in the solution including itself. I've therefore updated this project to get all its required dependencies from NuGet using `PackageReference` as it already seems to have been migrated to use this method (`packages.config` seemed redundant so I've deleted it).

Reply:
👍 this sounds good to me. 

To the unit test part:
Here is one test:
```
		/// <summary>
		/// Makes sure Agent.IsConfigured only returns true after Setup is called
		/// </summary>
		[Fact]
		public void IsConfigured()
		{
			Agent.IsConfigured.Should().BeFalse();

			Agent.Setup(new AgentComponents());
			Agent.IsConfigured.Should().BeTrue();
		}
```

You can add this to the `BasicAgentTests.cs` [here](https://github.com/elastic/apm-agent-dotnet/blob/master/test/Elastic.Apm.Tests/BasicAgentTests.cs#L36).

Reply:
Thanks. Have added this unit test in 28aa9a2.